
The ME909u and ME209u datasheets claim IPv6, and it seems to mostly work.

commit 6b6419bf978e7e8a89ecde6335cef6e9d5607d0c
Author: pchunt <peter.hunt@opengear.com>
Date:   Wed Jun 7 15:41:00 2017 -0600

    OG-4434 OG-4636 ModemManager: allow IPv4v6 and IPv6 for Huawei modems
    
    OG-4639

Index: ModemManager/plugins/huawei/mm-broadband-bearer-huawei.c
===================================================================
--- ModemManager.orig/plugins/huawei/mm-broadband-bearer-huawei.c
+++ ModemManager/plugins/huawei/mm-broadband-bearer-huawei.c
@@ -362,15 +362,6 @@ connect_3gpp_context_step (GTask *task)
             g_free (ip_family_str);
         }
 
-        if (ip_family != MM_BEARER_IP_FAMILY_IPV4) {
-            g_task_return_new_error (task,
-                                     MM_CORE_ERROR,
-                                     MM_CORE_ERROR_UNSUPPORTED,
-                                     "Only IPv4 is supported by this modem");
-            g_object_unref (task);
-            return;
-        }
-
         /* Store the task */
         self->priv->connect_pending = task;
 
